# Thực hiện thống kê mô tả trên tập dữu liệu về phân loại chất lượng rượu đỏ.


## 1. Import thư viện và nạp dữ liệu vào notebook.


In [15]:
import numpy as np
import pandas as pd
from IPython import display
from scipy import stats
df_dataset=pd.read_csv('winequality-red.csv')
display.display(df_dataset.head())
print(df_dataset.dtypes)
print(df_dataset.shape)

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

fixed acidity           float64
volatile acidity        float64
citric acid             float64
residual sugar          float64
chlorides               float64
free sulfur dioxide     float64
total sulfur dioxide    float64
density                 float64
pH                      float64
sulphates               float64
alcohol                 float64
quality                   int64
dtype: object
(1599, 12)


**Nhận xét:**
+ Dữ liệu phân loại rượu đỏ gồm có 12 thuộc tính từ dữ liệu winequality-red. Trong đó, dựa trên 11 thuộc tính ban đầu để quyết định phân loại chất lượng rượu đỏ như thế nào, tốt hay kém chất lượng. 
+ Dữ liệu gồm các kiểu số
+ Có 1599 dữ liệu dòng, tương đương với bộ dữ liệu với 1599 input.

## 2. Tính giá trị trung bình của 'quality'.

In [16]:
df_mean=np.mean(df_dataset['quality'])
print(df_mean)

5.6360225140712945


## 3. Tính median

In [17]:
df_median=np.median(df_dataset['quality'])
print(df_median)

6.0


## 4. Tính mode

In [21]:
df_mode = stats.mode(df_dataset["quality"])
print(df_mode)

ModeResult(mode=np.int64(5), count=np.int64(681))


## 5. Tính variance

In [22]:
df_variance=np.var(df_dataset['quality'])
print(df_variance)

0.6517605398308234


## 6. Tính MAX,MIN

In [24]:
df_max=np.max(df_dataset['quality'])
df_min=np.min(df_dataset['quality'])
print(f'Max: {df_max}')
print(f'Min; {df_min}')

Max: 8
Min; 3


## 7. Tính phân vị thứ 60


In [26]:
df_percentile=np.percentile(df_dataset['quality'],60)
print(df_percentile)

6.0


## 8. Tính tứ phân vị thứ ba (Q3, 75th percentile)

In [27]:
data_quartile = np.quantile(df_dataset['quality'], 0.75)
print(data_quartile)

6.0


## 9. Tính khoảng tứ phân vị (IQR)

In [29]:
data_IQR = stats.iqr(df_dataset["quality"])
print(data_IQR)

1.0
